modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -909,7 +910,26 @@ private ObjectField _addObjectField(
 				ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
 
 			if (localized) {
-				dbTableName = objectDefinition.getLocalizationDBTableName();
+				DynamicObjectDefinitionLocalizationTable
+					dynamicObjectDefinitionLocalizationTable =
+						DynamicObjectDefinitionLocalizationTableFactory.create(
+							objectDefinition, this);
+
+				dbTableName =
+					dynamicObjectDefinitionLocalizationTable.getTableName();
+
+				List<ObjectField> objectFields =
+					dynamicObjectDefinitionLocalizationTable.getObjectFields();
+
+				if (objectFields.size() == 1) {
+					runSQL("DROP_TABLE_IF_EXISTS(" + dbTableName + ")");
+
+					runSQL(
+						dynamicObjectDefinitionLocalizationTable.
+							getCreateTableSQL());
+
+					return objectField;
+				}
```

Comment:
I didn't understand:
1. You create the DynamicObjectDefinitionLocalizationTable.
2. Then, you're dropping > creating it again > adding no field (column)

We can talk in person so I can understand better.